odl/tomo/data/mrc2014.py

```diff
@@ -0,0 +1,846 @@
++---------+--------+----------+--------+-------------------------------+
+|57-256   |225-1024|String(80)|LABEL   |10 80-character text labels    |
++---------+--------+----------+--------+-------------------------------+
+"""
+
+
+def print_mrc2014_spec():
+    """Print the MRC2014 specification table.
+
+    The specification table is as follows:
+    """
+    print(MRC_2014_SPEC_TABLE)
+
+print_mrc2014_spec.__doc__ += MRC_2014_SPEC_TABLE
+
+
+MRC_HEADER_SIZE = 1024
+
+MRC_SPEC_KEYS = {
+    'id': 'Long word',
+    'byte_range': 'Byte',
+    'dtype': 'Data type',
+    'name': 'Name',
+    'description': 'Description'}
+
+MRC_DTYPE_TO_NPY_DTYPE = {
+    'Float32': np.dtype('float32'),
+    'Int32': np.dtype('int32'),
+    'String': np.dtype('S1')}
+
+MRC_MODE_TO_NPY_DTYPE = {
+    0: np.dtype('int8'),
+    1: np.dtype('int16'),
+    2: np.dtype('float32'),
+    6: np.dtype('uint16')}
+NPY_DTYPE_TO_MRC_MODE = {v: k for k, v in MRC_MODE_TO_NPY_DTYPE.items()}
+
+ANGSTROM_IN_METERS = 1e-10
+
+
+class FileReaderMRC(FileReaderRawBinaryWithHeader):
+
+    """Reader for the MRC file format.
+
+    By default, the MRC2014 format is used, see `print_mrc_2014_spec` for
+    details. See also [Che+2015]_ or the `explanations on the CCP4 homepage
+    <http://www.ccpem.ac.uk/mrc_format/mrc2014.php>`_ for the
+    text of the specification.
+
+    References
+    ----------
+    [Che+2015] Cheng, A et al. *MRC2014: Extensions to the MRC format header
+    for electron cryo-microscopy and tomography*. Journal of Structural
+    Biology, 129 (2015), pp 146--150.
+    """
+
+    def __init__(self, file, header_fields=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        file : file-like or str
+            Stream or filename from which to read the data. The stream
+            is allowed to be already opened in ``'rb'`` mode.
+        header_fields : sequence of dicts, optional
+            Definition of the fields in the header (per row), each
+            containing key-value pairs for the following keys:
+
+            - ``'name'`` : Label for the field.
+            - ``'offset'`` : Start of the field in bytes.
+            - ``'size'`` : Size of the field in bytes.
+            - ``'dtype'`` : Data type in Numpy- or Numpy-readable format.
+            - ``'dshape'`` (optional) : The array of values is reshaped to
+              this shape.
+            - ``'description'`` (optional) : A human-readable description
+              of the field.
+
+            For the default ``None``, the MRC2014 format is used, see
+            `print_mrc2014_spec`.
+        """
+        if header_fields is None:
+            header_fields = header_fields_from_table(
+                spec_table=MRC_2014_SPEC_TABLE,
+                keys=MRC_SPEC_KEYS,
+                dtype_map=MRC_DTYPE_TO_NPY_DTYPE)
+
+        super().__init__(file, header_fields)
+
+    print_mrc2014_spec = staticmethod(print_mrc2014_spec)
```

Comment:
This preserves the docstring. It's equivalent to decorating an inline function definition with `@staticmethod`.